src/webportal/src/app/job-submission/components/data/data-component.jsx

```diff
@@ -46,10 +46,16 @@ function reducer(state, action) {
 }
 
 export const DataComponent = React.memo((props) => {
-  // add pylonAddress to .env for local debug
-  const hdfsHost = config.hasOwnProperty('pylonAddress') ? getHostNameFromUrl(config.pylonAddress) : window.location.hostname;
+  const envsubRegex = /^\${.*}$/;
+  let hdfsHost;
+  if (!config.pylonAddress || envsubRegex.test(config.pylonAddress)) {
+    hdfsHost = window.location.hostname;
+  } else {
+    // add pylonAddress to .env for local debug
+    hdfsHost = getHostNameFromUrl(config.pylonAddress);
+  }
   // TODO: add a judgement whether pylon is ready
-  const hdfsClient = new WebHDFSClient(hdfsHost);
+  const hdfsClient = new WebHDFSClient(hdfsHost, undefined, undefined, 80, '/webhdfs/api/v1');
```

Comment:
Hmm, here I mean if we set customized hdfsHost for local debug, the hdfsClient may need to be created like:
```js
new WebHDFSClient(hdfsHost, undefined, undefined, 50700, '/webhdfs/v1');
```
or
```js
new WebHDFSClient(hdfsHost, undefined, undefined, port, '/webhdfs/v1');
```

Reply:
you mean not use pylon?